CacheMeIfYouCan/CacheMeIfYouCan/Protocols/DataConvertible.swift

```diff
@@ -25,3 +25,64 @@ public protocol DataConvertible {
     /// - Parameter data: The data to attempt to turn into the conforming type.
     init?(data: Data)
 }
+
+public extension Array where Element: DataConvertible {
+    
+    var toData: Data? {
+        let datas = self.compactMap { $0.toData }
+        if datas.isEmpty {
+            return nil
+        }
+        
+        return Data.mashTogether(array: datas)
+    }
+    
+    static func fromData(_ data: Data) -> [Element]? {
+        let datas = data.unmash
+        if datas.isEmpty {
+            return nil
+        }
+        
+        return datas.compactMap { Element(data: $0) }
+    }
+}
+
+extension Data {
+    
+    private static var separatorData: Data {
+        return "====== CACHE ME IF YOU CAN SEPARATOR =====".data(using: .utf8)!
+    }
+    
+    static func mashTogether(array: [Data]) -> Data {
+        var data = Data()
+        for (index, item) in array.enumerated() {
+            data.append(item)
+            if index != (array.count - 1) {
+                data.append(Data.separatorData)
```

Comment:
This is deep in your library away from your public funcs so it's not a crisis but encoding arrays of data with custom separators is not pretty.

I have been thinking of some alternatives --

Would it be possible to base64 encode each `item`, and then convert the array into a JSON array.

So you'd end up with the data structure that looks something like this on the way out of `mashTogether`:

```
[dGhlIGNha2U=, aXMgYQ==, bGll]
```

Each item is comma separated, so there's no magic separator, and it's easy to see how the items get serialised and deserialised.

Reply:
Yeah i agree it's not pretty - i think the only place where we're really using it is for things which are `DataConvertible` but which do *not* conform to `DataConvertibleCoadble`, which just encodes them into a JSON array and then writes them to disk. 

Might not even have to do it as a JSON array though - could just write the base 64 encoded data as newline separated strings. I'd have to look at how that affects performance on arrays of big chunks of data like images though.